core/iwasm/aot/aot_loader.c

```diff
@@ -2891,45 +2979,113 @@ aot_load_from_comp_data(AOTCompData *comp_data, AOTCompContext *comp_ctx,
 
     /* Allocate memory for function pointers */
     size = (uint64)module->func_count * sizeof(void *);
+    size += module->func_count;
     if (size > 0
         && !(module->func_ptrs =
                  loader_malloc(size, error_buf, error_buf_size))) {
         goto fail2;
     }
+    if (size > 0) {
+        module->func_ptrs_compiled =
+            (bool *)((uint8 *)module->func_ptrs
+                     + module->func_count * sizeof(void *));
+    }
 
-    /* fill in func_ptrs[] */
-    LOG_VERBOSE("to fill in func_ptrs ...");
-#if WASM_ENABLE_LAZY_JIT == 0
-    bh_print_time("Begin to do eager compilation...");
-#endif
-    for (i = 0; i < module->func_count; i++) {
-        LLVMErrorRef error;
-        LLVMOrcJITTargetAddress func_addr;
-        char func_name[32] = { 0 };
-
-        snprintf(func_name, sizeof(func_name), "%s%d", AOT_FUNC_PREFIX, i);
-#if WASM_ENABLE_LAZY_JIT != 0
-        error =
-            LLVMOrcLLLazyJITLookup(comp_ctx->orc_jit, &func_addr, func_name);
-#else
-        error = LLVMOrcLLJITLookup(comp_ctx->orc_jit, &func_addr, func_name);
-#endif
-        if (error != LLVMErrorSuccess) {
-            char *err_msg = LLVMGetErrorMessage(error);
-            set_error_buf_v(error_buf, error_buf_size,
-                            "failed to compile orc jit function: %s", err_msg);
-            LLVMDisposeErrorMessage(err_msg);
+    bh_print_time("Begin to lookup jit functions");
```

Comment:
how about "Start to do compilation in background"

Reply:
I did some test, it reduces port of total lookup time, had better add it to startup as quick as we can.